tests/baselines/reference/decoratorOnClassMethodThisParameter.types

```diff
@@ -0,0 +1,16 @@
+=== tests/cases/conformance/decorators/class/method/parameter/decoratorOnClassMethodThisParameter.ts ===
+declare function dec(target: Object, propertyKey: string | symbol, parameterIndex: number): void;
+>dec : (target: Object, propertyKey: string | symbol, parameterIndex: number) => void
+>target : Object
+>propertyKey : string | symbol
+>parameterIndex : number
+
+class C {
+>C : C
+
+    method(@dec this: C) {}
+>method : (: any, this: C) => void

```

Comment:
this type seems incorrectly printed?

Reply:
Are decorators on `this` parameters even a thing? O.o

Reply:
That's because this is a parse error (i.e. `@dec this` fails because `this` isn't an identifier), so you end up with a missing identifier.

Reply:
No, decorators on `this` parameters were never valid, but we did not have a test that reflects this. I opted to add one while investigating this fix

Reply:
Huh. Neat.